STYLE.md

```diff
@@ -52,6 +52,8 @@
   exceptions such as `main()` functions. When code cannot be placed inside the
   Envoy namespace there should be a comment of the form `// NOLINT(namespace-envoy)` at
   the top of the file.
+* If a public method is only intended to be called from test code then it should
+  have a name that ends in ForTest() such as aMethodForTest().
```

Comment:
I don't really have a strong opinion either way. My point mainly is that I think in practice it is actually quite rare that one needs to add a public method just for testing or friend the test code. Tests can frequently be written without doing any of that which IMO makes the tests better as the don't see into the inner details of the code under test. I would point that out, but I don't feel very strongly about it either way.